inkscape: 1.0.2 -> 1.1

jljusten opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change
Upstream update.

https://inkscape.org/news/2021/05/24/welcome-inkscape-11/
https://media.inkscape.org/media/doc/release_notes/1.1/Inkscape_1.1.html

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

jtojnar committed:
lib2geom: init at 1.0.0

jljusten committed:
inkscape: 1.0.2 -> 1.1

Reworks:
 * Remove fetchpatch
 * Remove gdl (jtojnar)
 * gtkspell -> gspell (jtojnar)
 * Add pillow to fix jpg export (jtojnar)
 * Remove double-conversion, add substituteInPlace (jtojnar)

r-rmcgibbo commented:
Result of `nixpkgs-review pr 124333` at a327d40b run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages marked as broken and skipped:</summary>
  <ul>
    <li>gnome.gnome-documents</li>
    <li>spring</li>
    <li>springLobby</li>
  </ul>
</details>
<details>
  <summary>47 packages skipped due to time constraints:</summary>
  <ul>
    <li>adapta-gtk-theme</li>
    <li>arc-theme</li>
    <li>arx-libertatis</li>
    <li>asciidoc-full</li>
    <li>asciidoc-full-with-plugins</li>
    <li>bibata-cursors</li>
    <li>bibata-cursors-translucent</li>
    <li>bibata-extra-cursors</li>
    <li>btrbk</li>
    <li>capitaine-cursors</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>inkscape</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `fetchpatch`.
Near pkgs/applications/graphics/inkscape/default.nix:8:3:
```
  |
8 | , fetchpatch
  |   ^
```
</li>
  </ul>
</details>

jtojnar reviewed (changes requested):
Some dependencies changed:

- gdl was replaced by new internal library
- gtkspell was replaced by gspell

jtojnar commented:
Exporting as JPEG fails with:

```
Traceback (most recent call last):
  File "raster_output_jpg.py", line 23, in <module>
    JpegOutput().run()
  File "/nix/store/gqxcnl2k0v4bm0f0pbyr8a6nx2zv44bf-inkscape-1.1/share/inkscape/extensions/inkex/base.py", line 130, in run
    self.load_raw()
  File "/nix/store/gqxcnl2k0v4bm0f0pbyr8a6nx2zv44bf-inkscape-1.1/share/inkscape/extensions/inkex/base.py", line 143, in load_raw
    document = self.load(self.file_io)
  File "/nix/store/gqxcnl2k0v4bm0f0pbyr8a6nx2zv44bf-inkscape-1.1/share/inkscape/extensions/inkex/extensions.py", line 73, in load
    from PIL import Image
ModuleNotFoundError: No module named 'PIL'
```

jtojnar commented:
Also lib2geom package can finally be used (needs to be updated first):


https://media.inkscape.org/media/doc/release_notes/1.1/Inkscape_1.1.html#Important_changes_for_packagers

Feel free to pick it from https://github.com/NixOS/nixpkgs/pull/50286

jljusten committed:
lib2geom: update 2020-03-12 -> 1.1

Reworks:
 * lgpl21 => lgpl21Only (jtojnar)
 * stdenv.lib => lib (jtojnar)
 * Use 1.1 tagged version rather than sha1 from inkscape 1.1 (jtojnar)
 * BUILD_SHARED_LIBS => 2GEOM_BUILD_SHARED (jtojnar)
 * Drop cmakeBuildType (jtojnar)

jljusten commented:
@jtojnar: Thanks for all the tips!

I removed gdl, added pillow for jpg export, and changed from gtkspell to gspell.

I tried to test spell check, but I didn't get the right dictionary installed yet, so it's not been verified.

I also pulled in your lib2geom patch, and updated it to the version used in inkscape 1.1. I tested that lib2geom built after this update. Unfortunately, when I tried to add it as a build dep for inkscape, I start seeing undefined references for double_conversion. Maybe you have a suggestion for this?

jtojnar commented on pkgs/development/libraries/lib2geom/default.nix:
```diff
@@ -0,0 +1,61 @@
+}:
+
+stdenv.mkDerivation rec {
+  pname = "lib2geom-unstable";
+  version = "2021-01-07";
```
Asked upstream to create a tag: https://gitlab.com/inkscape/lib2geom/-/merge_requests/48#note_584705175

SuperSandro2000 replied:
```suggestion
  pname = "lib2geom";
  version = "unstable-2021-01-07";
```

jtojnar commented on pkgs/development/libraries/lib2geom/default.nix:
```diff
@@ -0,0 +1,61 @@
+  meta = with stdenv.lib; {
+    description = "Easy to use 2D geometry library in C++";
+    homepage = "https://gitlab.com/inkscape/lib2geom";
+    license = [ licenses.lgpl21 licenses.mpl11 ];
```
The license is now called `lgpl21Only`.

SuperSandro2000 replied:
```suggestion
    license = with licenses; [ lgpl21Only mpl11 ];
```

jtojnar commented:
I guess it has something to do with the fact that it is built as static library, needs "-D2GEOM_BUILD_SHARED=ON"
.

jtojnar commented:
Also something like this:

```patch
--- a/pkgs/applications/graphics/inkscape/default.nix
+++ b/pkgs/applications/graphics/inkscape/default.nix
@@ -3,7 +3,6 @@
 , boost
 , cairo
 , cmake
-, double-conversion
 , fetchurl
 , gettext
 , ghostscript
@@ -21,6 +20,7 @@
 , libexif
 , libpng
 , librevenge
+, lib2geom
 , librsvg
 , libsigcxx
 , libsoup
@@ -83,6 +83,10 @@ stdenv.mkDerivation rec {
       --replace "call('ps2pdf'" "call('${ghostscript}/bin/ps2pdf'"
     patchShebangs share/templates
     patchShebangs man/fix-roff-punct
+
+    # double-conversion is a dependency of 2geom
+    substituteInPlace CMakeScripts/DefineDependsandFlags.cmake \
+      --replace 'find_package(DoubleConversion REQUIRED)' ""
   '';
 
   nativeBuildInputs = [
@@ -101,7 +105,6 @@ stdenv.mkDerivation rec {
   buildInputs = [
     boehmgc
     boost
-    double-conversion
     gettext
     glib
     glib-networking
@@ -114,6 +117,7 @@ stdenv.mkDerivation rec {
     libexif
     libpng
     librevenge
+    lib2geom
     librsvg # for loading icons
     libsigcxx
     libsoup

```

SuperSandro2000 commented on pkgs/development/libraries/lib2geom/default.nix:
```diff
@@ -0,0 +1,61 @@
+
+  doCheck = true;
+
+  meta = with stdenv.lib; {
```
```suggestion
  meta = with lib; {
```

jljusten commented:
@jtojnar: Yep, your suggested changes allow the lib2geom link to work. Thanks!

Hopefully I addressed all the other suggestions you made too.

I did have a slight concern about the lib2geom version, as I mentioned here: https://gitlab.com/inkscape/lib2geom/-/merge_requests/48#note_584835008

Nevertheless, I upgraded lib2geom to the 1.1 tagged version, even though it doesn't 100% match what the inkscape 1.1 used.

jtojnar commented on pkgs/development/libraries/lib2geom/default.nix:
```diff
@@ -0,0 +1,62 @@
+    gtest
+  ];
+
+  cmakeBuildType = "RelWithDebugInfo"; # needed to keep assertions for tests working
```
This does not seem to be necessary either.

jtojnar commented on pkgs/applications/graphics/inkscape/default.nix:
```diff
@@ -91,6 +83,10 @@ stdenv.mkDerivation rec {
     patchShebangs share/templates
     patchShebangs man/fix-roff-punct
+
+    # double-conversion is a dependency of 2geom
```
I have opened an upstream bug, could you please add a link to the comment? https://gitlab.com/inkscape/inkscape/-/issues/2508

jtojnar commented:
Great job. Thanks.

jljusten commented:
@jtojnar: Thanks for all the help with my first nixpkgs change. :)
